src/shared/components/palette/base-field/field-read.component.ts

```diff
@@ -19,6 +21,12 @@ export class FieldReadComponent extends AbstractFieldReadComponent implements On
   @Input()
   withLabel = false;
 
+  @Input()
```

Comment:
any field that is inject by this class to the various field read components, (`component.instance['caseReference'] = this.caseReference;`), is better declared on the abstract class AbstractFieldReadComponent. If you look at the fields that are currently injected, i.e. `caseField`, `caseReference`, `context`, these are all declared once on `AbstractFieldReadComponent`

Reply:
On another note, can I know why these two fields have been added to this component?

Reply:
Those were added because ccdConditionalShow and ccdLabelSubstitutor needs them. So if you want to use show and hide or markdown functionality in any template you have to have this data present.

Reply:
Ok so in that case I think we should try to find a better approach to get the case fields and form group down to the ccdConditionalShow and ccdLabelSubstitutor, where they will be actually used. Because adding these fields to the read and write components just to propagate them down could be avoided. These fields does not really belong to these classes. Let's see if we can find a better way to do that

Reply:
@rafalkalita I gave it a tought but can't find a better solution at the moment. If something comes to my mind later we can refactor as part of a new story, so please ignore the last comment above. But we could move those fields to the abstract classes. That would remove some duplication